src/vm/types/mod.rs

```diff
@@ -393,12 +712,13 @@ impl fmt::Display for Value {
             Value::Int(int) => write!(f, "{}", int),
             Value::UInt(int) => write!(f, "u{}", int),
             Value::Bool(boolean) => write!(f, "{}", boolean),
-            Value::Buffer(vec_bytes) => write!(f, "0x{}", &vec_bytes),
             Value::Tuple(data) => write!(f, "{}", data),
             Value::Principal(principal_data) => write!(f, "{}", principal_data),
             Value::Optional(opt_data) => write!(f, "{}", opt_data),
             Value::Response(res_data) => write!(f, "{}", res_data),
-            Value::List(list_data) => {
+            Value::Sequence(SequenceData::Buffer(vec_bytes)) => write!(f, "0x{}", &vec_bytes),
+            Value::Sequence(SequenceData::String(string)) => write!(f, "{}", string),
```

Comment:
I think the display formats for the string types should include their double quotes and escape chars, e.g.

String(ASCII) => "abc\"d"
String(UTF8) => u"abc\"d"

So that the produced string is Clarity code. At a minimum, I think it should include quotes so that the string "(ok 1)" can be differentiated from the display for `Value::Some(1)`. What do you think?

Reply:
Yeah, I've been doing some back and forth on this, my initial approach was including some double quotes, and got rid of it at some point. Will revisit.

Reply:
> I think the display formats for the string types should include their double quotes and escape chars

This in reference to using `print` to display strings, right?

> So that the produced string is Clarity code.

With the goal of being able to use the Clarity parser to read back literals?

Reply:
>> I think the display formats for the string types should include their double quotes and escape chars
> This in reference to using print to display strings, right?

This would affect how `print` works, but also other outputs from the stacks-blockchain binary, e.g., logging outputs, outputs from the `clarity-cli` bin, and the repl. 

>> So that the produced string is Clarity code.
> With the goal of being able to use the Clarity parser to read back literals?

No, not necessarily. It can just make for a nicer developer experience for the printed display of a value to also be code that would yield that value. Many repls provide display formats that have that property.

Reply:
> It can just make for a nicer developer experience for the printed display of a value to also be code that would yield that value.

Yes! For many good reasons.  

> This would affect how print works, but also other outputs from the stacks-blockchain binary

There is indeed an important distinction between output that follows the syntax of the language, and output that are just meant to be read as plain text. Lisp languages typically have different print/formatting functions for these different needs. Maybe we should have something similar, like say a `format` function for the latter case, leaving the `print` function to output in Clarity syntax.

Reply:
Addressed with https://github.com/blockstack/stacks-blockchain/pull/1779/commits/9689dd679cff117cd18336036a5d44b5de611339